luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/wysiwyg_tool.rs

```diff
@@ -45,6 +46,17 @@ impl Component for WysiwygTool<'_> {
             on_event: Box::new(|event| on_event(Event::Resizer { event })),
         });
 
+        ctx.component(Rotator {
+            rect: graphic_dest_rect,
+            dragging_context: if let Some(Dragging::Rotator { context }) = self.dragging.as_ref() {
+                Some(*context)
+            } else {
+                None
+            },
```

Comment:
use map please

Reply:
I don't think map is available here because we need to check `Dragging::Rotator`.

You mean like this?
```rust
dragging_context: self.dragging.as_ref().and_then(|dragging| {
    if let Dragging::Rotator { context } = dragging {
        return Some(*context);
    }
    None
}),
```

Reply:
AH it's enum! okokokokokok Nevermind